Source/BugsnagStacktrace.m

```diff
@@ -0,0 +1,46 @@
+//
+//  BugsnagStacktrace.m
+//  Bugsnag
+//
+//  Created by Jamie Lynch on 06/04/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import "BugsnagStacktrace.h"
+#import "BugsnagStackframe.h"
+#import "BugsnagCollections.h"
+#import "BugsnagKeys.h"
+
+@interface BugsnagStackframe ()
++ (BugsnagStackframe *)frameFromDict:(NSDictionary *)dict
+                          withImages:(NSArray *)binaryImages;
+- (NSDictionary *)toDict;
+@end
+
+@implementation BugsnagStacktrace
+
+- (instancetype)initWithTrace:(NSArray<NSDictionary *> *)trace
+                 binaryImages:(NSArray<NSDictionary *> *)binaryImages {
+    if (self = [super init]) {
+        self.trace = [NSMutableArray new];
+
+        for (NSDictionary *obj in trace) {
+            BugsnagStackframe *frame = [BugsnagStackframe frameFromDict:obj withImages:binaryImages];
+
+            if (frame != nil && [self.trace count] < 200) {
+                [self.trace addObject:frame];
+            }
```

Comment:
Is it worth adding logging here if we're discarding frames?

Reply:
I believe the current behaviour is to discard at least one frame from most KSCrash reports, which is done silently. Presumably the reason behind this is that otherwise we'd log a message out for each event that is captured.

Reply:
My thinking was that exceeding 200 frames is an unusual event, and may in itself point to a situation that needs investigated during development, so it was logging (once) when the upper limit is exceeded, not every single discard.

Reply:
Agreed after IRL discussion to raise this as a point with the spec, and address separately if any action required

Reply:
It's been agreed that this will be looked at at a later date.